### PR TITLE
Fix json collector decorator and switch json parsing library to orjson

### PR DESCRIPTION
**Pull request by aspacca**
Bug/Enhancement
## What does this PR do?
Fix a bug related to event list from field expander where, if json content type was single and spanning more than 1000 lines the circuit breaker of the json decorator will kick in

It also switch orjson parsing library over ujson one

## Why is it important?
A regression was introduced on json decorator while refactored with #544 
Also switching from usjon to orjson improves performances

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: handlers/aws/utils.py [hunk 1]
```diff
@@ -313,8 +313,11 @@ def get_trigger_type_and_config_source(event: dict[str, Any]) -> tuple[str, str]
                 and "eventSource" in body["Records"][0]
             ):
                 event_source = body["Records"][0]["eventSource"]
+                if event_source not in _available_triggers:
+                    raise Exception
```
Comment (girodav) [inline handlers/aws/utils.py:317]: I would add an exception message of some kind here

File: storage/decorator.py [hunk 2]
```diff
@@ -236,6 +236,21 @@ def wrapper(
 
             yield line, _, _, newline, None
 
+    def _collect_single(iterator: StorageDecoratorIterator) -> StorageDecoratorIterator:
+        single: list[tuple[Union[StorageReader, bytes], int, int, bytes]] = list(
+            [
+                (data, starting_offset, ending_offset, newline)
+                for data, starting_offset, ending_offset, newline, _ in iterator
+            ]
+        )
```
Comment (girodav) [inline storage/decorator.py:245]: I find this DS a bit hard to digest. Maybe a comment with an example entry could help

File: storage/decorator.py [hunk 3]
```diff
@@ -255,43 +270,36 @@ def wrapper(
...
+            if storage.json_content_type == "single":
+                iterator = _collect_single(iterator=iterator)
+                json_collector_state.is_a_json_object = True
+                json_collector_state.has_an_object_start = True
+
+            for data, starting_offset, ending_offset, newline, _ in iterator:
+                assert isinstance(data, bytes)
+
+                # if it's not a json object we can just forward the content by lines
+                if not json_collector_state.has_an_object_start:
```
Comment (girodav) [inline storage/decorator.py:284]: This comment does not seem to be consistent with the if statement below. Should this use `json_collector_state.is_a_json_object` instead?

**Review by girodav — COMMENTED** [review]
Thanks for this. I wonder if we should also make the PR title more descriptive. Pre-release sounds a bit generic.

File: handlers/aws/utils.py (hunk 1 above)
Comment (aspacca) [reply handlers/aws/utils.py:317]: I added a message only where the exceptions will bubble up to the function callar

**Comment by aspacca** [comment]
@girodav addressed your comment, please review :)

**Review by girodav — APPROVED** [review]

**Comment by aspacca** [comment]
@girodav , I fixed a couple of comments and review status was dismissed

feel free to re-approve and merge, thanks :)

**Review by girodav — APPROVED** [review]

**Pull request merged**